test/e2e/network/service.go

```diff
@@ -1852,8 +1853,112 @@ var _ = SIGDescribe("Services", func() {
 		}
 	})
 
+	// This test verifies if service load balancer cleanup finalizer can be removed
+	// when feature gate isn't enabled on the cluster.
+	// This ensures downgrading from higher version cluster will not break LoadBalancer
+	// type service.
+	ginkgo.It("should remove load balancer cleanup finalizer when service is deleted [Slow]", func() {
+		jig := framework.NewServiceTestJig(cs, "lb-remove-finalizer")
+
+		ginkgo.By("Create load balancer service")
+		svc := jig.CreateTCPServiceOrFail(f.Namespace.Name, func(svc *v1.Service) {
+			svc.Spec.Type = v1.ServiceTypeLoadBalancer
+		})
+
+		defer func() {
+			waitForServiceDeletedWithFinalizer(cs, svc.Namespace, svc.Name)
```

Comment:
Is there a real LB to verify against? Maybe other tests verify this enough

Reply:
Right, @prameshj raised similar point offline. We have other LB tests that actually verify against real LBs (e.g. checking the actual LB resources and make sure they are cleaned up). But it is valid point that it would be better to check it here.  https://github.com/kubernetes/kubernetes/issues/78549 will track this work as followup if that is ok.